File: retos/estructuras-datos/matrices/agentes_secretos.js
const str = "OSELNKTEPT-ANUEEREE";
const filas = str.length / 4;
const columnas = 4;
const object = { length: Math.ceil(filas) }
const matriz = Array.from(object, () => Array(columnas).fill(0));

let index = 0;



// Rellenar por columnas
for (let j = 0; j < columnas; j++) {
  for (let i = 0; i < filas; i++) {
    matriz[i][j] = str[index++] ?? 'X';
  }
}

console.table(matriz);
let mensaje = ""

//leer mensaje
for (let i =0; i<filas; i++){
    for (let j=0; j<columnas; j++){
        mensaje += matriz[i][j]
    }
}

console.log(mensaje.split('X').join(''));

File: retos/estructurasDatos/matrices/agentes-secretos/agentesSecretos.cpp
#include <iostream>
#include <string>
#include <vector>
#include <cmath>

int main() {
    std::string str = "OSELNKTEPT-ANUEEREE";
    int col = 4;
    int filas = static_cast<int>(std::ceil(str.length() / 4.0));
    int index = 0;
    std::string mensaje = "";

    // Matriz dinámica filas x col de strings
    std::vector<std::vector<std::string>> matriz(filas, std::vector<std::string>(col));

    // Llenar la matriz por columnas
    for (int j = 0; j < col; j++) {
        for (int i = 0; i < filas; i++) {
            matriz[i][j] = (index < (int)str.length()) ? str.substr(index++, 1) : "X";
        }
    }

    // Construir el mensaje leyendo fila por fila
    for (int i = 0; i < filas; i++) {
        for (int j = 0; j < col; j++) {
            mensaje += matriz[i][j];
        }
    }

    std::cout << mensaje << std::endl;

    return 0;
}
